tools/docker/builder/openwrt/build.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/sh -e
+
+scriptdir="$(cd "${0%/*}"; pwd)"
+
+if [ ! -f "$1" ] ; then
+    echo "Please specify a file containing the build args as the
+    first argument"
+    exit 1
+fi
+
+. "$(realpath "$1")"
+
+if [ -z "$OPENWRT_REPOSITORY" ] ; then
+    OPENWRT_REPOSITORY=https://git.prpl.dev/prplwrt/prplwrt.git
+    echo "OPENWRT_REPOSITORY not set, using default value $OPENWRT_REPOSITORY"
+
+fi
+
+if [ -z "$OPENWRT_VERSION" ] ; then
+    OPENWRT_VERSION=9d2efd
+    echo "OPENWRT_VERSION not set, using default value $OPENWRT_VERSION"
+fi
+
+if [ -z "$TARGET" ] ; then
+    TARGET=mvebu
+    echo "TARGET not set, using default value $TARGET"
+fi
+
+if [ -z "$TARGET_PLATFORM_TYPE" ] ; then
+    TARGET_PLATFORM_TYPE=turris-omnia
+    echo "TARGET_PLATFORM_TYPE not set, using default value $TARGET_PLATFORM_TYPE"
+fi
+
+if [ -z "$USE_DWPAL" ] ; then
+    USE_DWPAL=false
+    echo "USE_DWPAL not set, using default value $USE_DWPAL"
+fi
+
+if [ -z "$PRPLMESH_VERSION" ] ; then
+    PRPLMESH_VERSION=82f1e83
+    echo "PRPLMESH_VERSION not set, using default value $PRPLMESH_VERSION"
+fi
+
+export OPENWRT_REPOSITORY
+export OPENWRT_VERSION
+export TARGET
+export TARGET_PLATFORM_TYPE
+export USE_DWPAL
+export PRPLMESH_VERSION
+
+image_tag="prplmesh-builder-${TARGET}:${OPENWRT_VERSION}"
+container_name="prplmesh-builder-$(date +%F_%H-%M-%S)"
+
+echo "Image will be tagged as $image_tag"
+echo "Container name will be $container_name"
+
+# We first need to build the corresponding images
+docker build --tag "$image_tag" \
+       --build-arg OPENWRT_REPOSITORY \
+       --build-arg OPENWRT_VERSION \
+       --build-arg TARGET \
+       --build-arg PRPL_FEED='https://git.prpl.dev/prplmesh/iwlwav.git' \
+       .
+
+# Next we'll build the specified prplMesh version and copy out the ipk
+docker run -i \
+       --name "$container_name" \
+       -e TARGET \
+       -e TARGET_PLATFORM_TYPE \
+       -e OPENWRT_VERSION \
+       -e USE_DWPAL \
+       -e PRPLMESH_VERSION \
```

Comment:
I went with [USE_SOURCE_DIR](https://openwrt.org/docs/guide-developer/packages#use_source_dir) since it doesn't require to commit changes in order to build. Since it requires a separate "prepare" step, we will have to document how to rebuild manually  somewhere.